src/hip/handlehip.cpp

```diff
@@ -484,18 +484,36 @@ Program Handle::LoadProgram(const std::string& program_name,
                             const std::string& kernel_src) const
 {
     this->impl->set_ctx();
+    std::string arch_name = this->GetTargetProperties().Name();
+
+    std::string orig_params = params; // make a copy for target ID fallback
 
     if(!miopen::EndsWith(program_name, ".mlir"))
-    {
-        params += " -mcpu=" + this->GetTargetProperties().Name();
-    }
+        params = params + " -mcpu=" + this->GetTargetProperties().Name();
 
     auto hsaco = miopen::LoadBinary(this->GetTargetProperties(),
                                     this->GetMaxComputeUnits(),
                                     program_name,
                                     params,
                                     is_kernel_str);
     if(hsaco.empty())
+    {
+        const auto arch_target_id = miopen::SplitDelim(arch_name, ':');
+        if(arch_target_id.size() > 1)
+        {
+            // The target name has target ID in there, fall back on the generic code object
```

Comment:
@JehandadKhan @junliume `TargetProperties` is intended to hold values of target properties: name (string) and boolean attributes like xnack, sramecc etc. Therefore `GetTargetProperties().Name()` do not contian any compiler-speicific info (unlike things like `LcOptionTargetStrings` which resides in hip_build_utils.hpp).

Therefore the change in `Handle::LoadProgram()` is useless and can be reverted.